qst/src/main/java/io/deephaven/qst/table/ParentsVisitor.java

```diff
@@ -24,78 +28,53 @@
      *
-     * @see #postOrderWalk(Iterable, Consumer, int)
-     */
-    public static Set<TableSpec> postOrder(Iterable<TableSpec> tables, int maxDepth) {
-        Set<TableSpec> set = new LinkedHashSet<>();
-        postOrderWalk(tables, set::add, maxDepth);
-        return set;
-    }
-
-    /**
-     * Walk the {@link TableSpec tables} in post-order with de-duplication.
+     * <p>
+     * Post-order means that for any given table, the table's dependencies will come before the
+     * table itself. There may be multiple valid post-orderings; callers should not rely on a
+     * specific post-ordering.
      *
      * @param tables the tables
-     * @param consumer the consumer
+     * @return the de-duplicated, post-order list
      */
-    public static void postOrderWalk(Iterable<TableSpec> tables, Consumer<TableSpec> consumer) {
-        postOrderWalk(tables, consumer, Integer.MAX_VALUE);
-    }
-
-    /**
-     * Walk the {@link TableSpec tables} post-order with de-duplication, up to a depth of
-     * {@code maxDepth}.
-     *
-     * @param tables the tables
-     * @param consumer the consumer
-     * @param maxDepth the maximum depth
-     */
-    public static void postOrderWalk(Iterable<TableSpec> tables, Consumer<TableSpec> consumer,
-        int maxDepth) {
-        Set<TableSpec> visited = new HashSet<>();
-        for (TableSpec table : tables) {
-            postOrderTraversal(visited, table, consumer, maxDepth);
-        }
-    }
-
-    private static void postOrderTraversal(Set<TableSpec> visited, TableSpec table,
-        Consumer<TableSpec> consumer, int maxDepth) {
-        // This method is much more efficient than trying to accomplish the same with
-        // Stream#distinct, since we
-        // can cut off the duplication at the highest-level table.
-        if (maxDepth < 0 || !visited.add(table)) {
-            return;
-        }
-        if (maxDepth > 0) {
-            try (Stream<TableSpec> stream = getParents(table)) {
-                Iterator<TableSpec> it = stream.iterator();
-                while (it.hasNext()) {
-                    postOrderTraversal(visited, it.next(), consumer, maxDepth - 1);
+    public static List<TableSpec> postOrderList(Iterable<TableSpec> tables) {
+        List<TableSpec> postOrder = new ArrayList<>(anyOrder(tables));
+        postOrder.sort(Comparator.comparingInt(TableSpec::depth));
+        return postOrder;
+    }
+
+    private static Set<TableSpec> anyOrder(Iterable<TableSpec> initialInputs) {
+        Set<TableSpec> output = new HashSet<>();
+        Queue<TableSpec> toProcess = new ArrayDeque<>();
+        for (TableSpec initialInput : initialInputs) {
+            toProcess.add(initialInput);
+            do {
+                final TableSpec table = toProcess.remove();
+                if (output.add(table)) {
+                    final Iterator<TableSpec> it = ParentsVisitor.getParents(table).iterator();
+                    while (it.hasNext()) {
```

Comment:
I think you're over-complicating this.
`ParentsVisitor.getParents(table).forEach(toProcess::add);`